chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -150,7 +150,8 @@ def self.unclaimed(node, include_mounted=false)
               %x{lsblk #{d.name.gsub(/!/, "/")} --noheadings --output MOUNTPOINT | grep -q -v ^$}
               next if $?.exitstatus == 0
             end
-            d.fixed and not d.claimed?
+            next unless d.fixed || d.multipath?
```

Comment:
No sorry, it's really not straightforward to me. (N.B. I'm talking about both these lines, not just one of them.)  I think I managed to understand it last time I looked at it, but looking at it again now, I feel just as confused as I did originally.  There are *four* different things being checked here:

- `d.fixed`
- `d.multipath?`
- `d.claimed?`
- `d.multipath_path?`

And on top of that we have a nice Boolean cocktail of `unless`, `||`, `&&`, and `!`.  So that creates a truth table with 2^4==16 different possibilities, which is not simple to understand.

Some questions which immediately spring to mind:

- Why are two of the conditions used by `next`, and the other two used to form the return value?
- Why are we checking four different things?  The only obvious one is `!d.claimed?` since that corresponds directly to the method name `unclaimed`.  But why are we skipping non-fixed devices, for example?

To be clear: I'm not saying there's anything wrong with the logic, just that it's hard to understand the way it's currently written.  If you really understand it then surely it should be a matter of 60 seconds to add a comment or two which would go a long way to helping others understand.

This is the kind of situation in which @cmurphy's [talk](https://youtu.be/pU-VasVPNAs) seems particularly relevant to me.  I regularly see patches submitted to lots of projects where the submitter assumes that the reviewers have all the necessary context and knowledge to be able to easily review the patch, but this is often not the case.